### DISCUSSION TITLE
What object-NodeId to use for method call

### DISCUSSION THREAD

**Discussion by hannosius** [eclipse-milo/milo#961] [General]
Hi,
I am currently trying to figure out how method calling is supposed to be done.

This is the tree of the server:
![grafik](https://user-images.githubusercontent.com/31645793/163203120-3ad3b357-49d7-4f1e-b4e7-7065c5b9297f.png)

The "Set output"-method takes one uint parameter indicating the number of the DigitalOutput you want to set.

I am trying to call this method with the following code:

	public boolean callMethod(OpcUaClient client) throws UaException, InterruptedException, ExecutionException {

		NodeId object = new NodeId(2, 15150); //???
		NodeId method = new NodeId(1, 60042);

		CallMethodRequest req = new CallMethodRequest(object, method,
				new Variant[] { new Variant(UInteger.valueOf(5)) });

		StatusCode status = client.call(req).get().getStatusCode();
		return status.isGood();
	}

This doesn't work. I am not really sure what NodeId to put in for the object-NodeId. 
So my guess is that I am using the wrong one there.
Can anyone tell me what I am doing wrong here and what NodeId to use for the objectid-parameter of the CallMethodRequest-constructor?

Thanks in advance for your help!

**Comment by kevinherron** [comment]
The Object NodeId should be that of the parent Object Node that has a reference to the Method Node, i.e. to call the `Set output` method in your screenshot the Object NodeId is going to be `ns=0;i=85` (`Identifiers.ObjectsFolder`).

**Comment by hannosius** [comment]
Hi Kevin,
Thank you very much for your quick response! 
I was actually sure that I had already tried that (ns=0;i=85).
But when I did now it worked. 
So thank you very much!
